torchci/clickhouse_queries/ttrs_percentiles/query.sql

```diff
@@ -71,58 +58,74 @@ pr_shas AS (
   WHERE
-    pr.closed_at IS NOT NULL -- Ensure the PR was actaully merged
-    AND 'Merged' IN (
-      SELECT
-        name
-      FROM
-        UNNEST(pr.labels)
-    )
+    pr.closed_at != '' -- Ensure the PR was actaully merged
+    AND label.name = 'Merged'
 ),
 -- Get all the workflows run against the PR and find the steps & stats we care about
 commit_job_durations AS (
   SELECT
-    s.pr_number,
-    j.steps,
-    js.name AS step_name,
-    js.conclusion AS step_conclusion,
-    PARSE_TIMESTAMP_ISO8601(js.completed_at) AS failure_time,
-    PARSE_TIMESTAMP_ISO8601(js.started_at) AS start_time,
+    s.pr_number as pr_number,
+    j.steps as steps,
     r.name AS workflow_name,
     j.name AS job_name,
     r.html_url AS workflow_url,
     -- for debugging
-    s.sha,
+    s.sha as sha,
     j.conclusion AS conclusion,
     j.conclusion = 'cancelled' AS was_cancelled,
     -- For convenience
-    j.run_attempt,
-    -- the attemp number this job was run on
+    j.run_attempt as run_attempt,
+    -- the attempt number this job was run on
     r.run_attempt AS total_attempts,
     r.id AS workflow_run_id,
-    s.url -- for debugging
+    s.url as url -- for debugging
   FROM
-    commons.workflow_job j
-    INNER JOIN merged_pr_shas s ON j.head_sha = s.sha HINT(join_strategy = lookup)
-    CROSS JOIN UNNEST (j.steps) js
-    INNER JOIN commons.workflow_run r ON j.run_id = r.id
+     default.workflow_job j final
+     JOIN merged_pr_shas s ON j.head_sha = s.sha
+     JOIN default.workflow_run r final ON j.run_id = r.id
   WHERE
-    1 = 1
-    AND r.name = :workflow -- Stick to pull workflows to reduce noise. Trendlines are the same within other workflows
+    r.name = {workflow: String} -- Stick to pull workflows to reduce noise. Trendlines are the same within other workflows
+    and j.id in (
+      select id from materialized_views.workflow_job_by_head_sha mv
+      where mv.head_sha in (select sha from merged_pr_shas)
+    )
     AND j.conclusion = 'failure' -- we just care about failed jobs
-    AND js.conclusion = 'failure'
     AND j.run_attempt = 1 -- only look at the first run attempt since reruns will either 1) succeed, so are irrelevant or 2) repro the failure, biasing our data
     and j.name NOT LIKE 'lintrunner%'
     and j.name NOT LIKE '%unstable%' -- The PR doesn't wait for unstable jobs, so they should be excluded when computing TTRS
-    and js.name LIKE 'Test%' -- Only consider test steps
-    ),
+  ),
+  commit_job_durations_steps as (
+      SELECT
+    j.pr_number,
+    js.'name' AS step_name,
+    js.'conclusion' AS step_conclusion,
+    js.'completed_at' AS failure_time,
+    js.'started_at' AS start_time,
+    j.workflow_name,
+    j.job_name,
+    j.workflow_url,
+    j.sha,
+    j.conclusion,
+    j.conclusion = 'cancelled' AS was_cancelled,
+    j.run_attempt,
+    j.total_attempts,
+    j.workflow_run_id,
+    j.url -- for debugging
+  FROM
+   commit_job_durations j
+    array JOIN j.steps as js
```

Comment:
Had to split out the array join because ch didn't like having array join with other joins (even though it was fine earlier?  might be that there are two joins in addition instead of just one)